aws/resource_aws_globalaccelerator_listener.go

```diff
@@ -0,0 +1,270 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/globalaccelerator"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsGlobalAcceleratorListener() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsGlobalAcceleratorListenerCreate,
+		Read:   resourceAwsGlobalAcceleratorListenerRead,
+		Update: resourceAwsGlobalAcceleratorListenerUpdate,
+		Delete: resourceAwsGlobalAcceleratorListenerDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"accelerator_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"client_affinity": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					globalaccelerator.ClientAffinityNone,
+					globalaccelerator.ClientAffinitySourceIp,
+				}, false),
+			},
+			"protocol": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					globalaccelerator.ProtocolTcp,
+					globalaccelerator.ProtocolUdp,
+				}, false),
+			},
+			"port_range": {
+				Type:     schema.TypeList,
```

Comment:
Does the ordering matter for these? If not, we should use `schema.TypeSet`. It would be nice to verify this either which way with an acceptance test that has multiple `port_range` configurations.